### ISSUE TITLE
[Request] upgrading clang to latest version

### ISSUE THREAD

**Issue by mast3rz3ro** [msys2/MSYS2-packages#4316]
Hi there,
Can you please add latest version of clang to msys2 packages, the latest version i seen at the moment is 11.x however mingw64 has clang version 17.x

I need latest clang to compile a program which depends on memmem function.

Before i open this issue i tried to compile clang but the compiling process can take a lot of time! (i was waiting for approximately 3 hours and only a 700 of 4000 units were compiled.)

Waiting for your anwsers
Thank you!

**Comment by mast3rz3ro** [comment]
[mmuetzel](https://github.com/mmuetzel) Is this mean no more new updates for clang/llvm ?

**Comment by mmuetzel** [comment]
IIUC, there never was a Clang compiler that targeted Cygwin. That cross-compiler fulfilled its intended use.
I'd guess there probably won't be an update to that cross-compiler. Maybe, there will be a "native" Cygwin Clang if LLVM should ever start to support that target.

Why don't you use the native compiler from the CLANG64 environment instead?

**Comment by MehdiChinoune** [comment]
MSYS2 environment is not intended to be used by itself, It's for providing tools to build MinGW packages.

**Comment by mast3rz3ro** [comment]
> Why don't you use the native compiler from the CLANG64 environment instead?

Because clang doesn't provide memmem function but the newlib at least it's has an accepted implementation.

> MSYS2 environment is not intended to be used by itself, It's for providing tools to build MinGW packages.

I just notice that early, recently i have started using the latest mingw/clang builds provided by [winlibs](https://winlibs.com/) at least i can place these builds manually into Msys2 without having any issues. In case you may wondering why i do this? because the last version of MSYS2 with Windows 7 support was "20221028" and i can't update the packages to the latest available normally (even if i do pacman -Fy) unless i do update the whole system core by following this guide [here](https://www.msys2.org/docs/updating/).

thanks you for the answers.

**Comment by mmuetzel** [comment]
> Why don't you use the native compiler from the CLANG64 environment instead?\n\nBecause clang doesn't provide memmem function but the newlib at least it's has an accepted implementation.

I'm not sure I understand correctly. Are you linking newlib into a binary that you are building with that cross-compiler?
That will likely cause issues down the line. Cygwin and native Windows binaries aren't compatible.

**Comment by mast3rz3ro** [comment]
> I'm not sure I understand correctly. Are you linking newlib into a binary that you are building with that cross-compiler?
No, I mean I'm required to use newlib because it's has an stable implementation of memmem function however the last clang version provides MSYS2 (with normal package name) is CLANG 11.0.0-5 but some of these source codes I'm compiling requires an updated/latest clang version (For example take look on [this](https://github.com/plooshi/KPlooshFinder/issues/2)).

> That will likely cause issues down the line. Cygwin and native Windows binaries aren't compatible.

I know that because I have faced some performance issues with some projects that I have compiled with newlib.

I always try to compile using mingw (for many reasons) but some times I'm enforced to use newlib for compiling some of glibc based projects.

Thank you, feel free to close this issue.
